diff -Naur -x .bzr -x CVS upstream/dhcp6c.8 upstream-cvs/dhcp6c.8
--- upstream/dhcp6c.8	2006-07-30 03:24:19.000000000 -0700
+++ upstream-cvs/dhcp6c.8	2006-12-06 09:01:45.000000000 -0800
@@ -74,7 +74,19 @@
 it prints the messages to standard error if this option is
 specified.
 .It Fl i
-Info-req mode.  In this mode, stateless DHCPv6 is executed with an automatic configuration, and the obtained info is written to stdout.  After this output, the program is terminated.  (suits for a use in shell-script etc)
+Info-req mode.  In this mode, stateless DHCPv6 is executed with the folllowing configuration, and the obtained info is written to stdout.  After this output,
+.Nm
+is terminated.  (suits for a use in shell-script etc).
+.Bd -literal -offset
+ interface (interface given in the argument) {
+	information-only;
+	script (a script which displays the received information to stdout);
+ };
+.Ed
+.Pp
+Since the configuration is internally generated, you cannot provide a configuration in this mode.  If you want to have different actions for the stateless DHCPv6 information, you should write an appropriate configuration and invoke
+.Nm
+without this option.
 .It Fl p Ar pid-file
 Use
 .Ar pid-file
diff -Naur -x .bzr -x CVS upstream/dhcp6relay.c upstream-cvs/dhcp6relay.c
--- upstream/dhcp6relay.c	2006-07-28 18:42:09.000000000 -0700
+++ upstream-cvs/dhcp6relay.c	2006-12-06 09:01:46.000000000 -0800
@@ -100,6 +100,7 @@
 static char *global_strings[] = {
 	/* "fec0::/10",	site-local unicast addresses were deprecated */
 	"2000::/3",
+	"FC00::/7",  /* Unique Local Address (RFC4193) */
 	NULL
 };
 
diff -Naur -x .bzr -x CVS upstream/dhcp6s.c upstream-cvs/dhcp6s.c
--- upstream/dhcp6s.c	2006-08-06 21:35:32.000000000 -0700
+++ upstream-cvs/dhcp6s.c	2006-11-18 13:18:22.000000000 -0800
@@ -279,6 +279,7 @@
 			break;
 		case 'P':
 			pid_file = optarg;
+			break;
 		default:
 			usage();
 			/* NOTREACHED */
@@ -2500,13 +2501,12 @@
 		    bia = TAILQ_NEXT(bia, link)) {
 			if ((v = dhcp6_find_listval(conflist,
 			    bia->type, &bia->uv, 0)) != NULL) {
-				found++;
 				TAILQ_REMOVE(conflist, v, link);
 				dhcp6_clear_listval(v);
 			}
 		}
 
-		return (found);
+		return (1);
 	}
 
 	/*
diff -Naur -x .bzr -x CVS upstream/Makefile.in upstream-cvs/Makefile.in
--- upstream/Makefile.in	2006-10-13 05:19:12.000000000 -0700
+++ upstream-cvs/Makefile.in	2006-11-18 13:18:22.000000000 -0800
@@ -38,7 +38,7 @@
 	-DLOCALDBDIR=\"${localdbdir}\"
 LDFLAGS=@LDFLAGS@
 LIBOBJS=@LIBOBJS@
-LIBS=	@LIBS@ -ly -ll
+LIBS=	@LIBS@ @LEXLIB@
 CC=	@CC@
 TARGET=	dhcp6c dhcp6s dhcp6relay dhcp6ctl
 
@@ -94,7 +94,7 @@
 	expand bootp-dhcp-parameters | perl gentab.pl > ianaopts.h
 
 install::
-	-mkdir -p $(sbindir) $(mandir)
+	-mkdir -p $(sbindir) $(mandir)/man5 $(mandir)/man8
 	$(INSTALL_PROGRAM) -s -o bin -g bin $(TARGET) $(sbindir)
 	$(INSTALL_DATA) -o bin -g bin dhcp6c.8 $(mandir)/man8
 	$(INSTALL_DATA) -o bin -g bin dhcp6s.8 $(mandir)/man8
