--- common.c	2007/06/14 05:18:29	1.11
+++ common.c	2007/07/22 06:59:23	1.12
@@ -663,8 +663,11 @@
 	prev = ntohq(prev);
 	current = ntohq(current);
 
-	/* we call the singular point guilty */
-        if (prev == (current ^ 8000000000000000ULL)) {
+	/*
+	 * we call the singular point guilty, since we cannot guess
+	 * whether the serial number is increasing or not.
+	 */
+        if (prev == (current ^ 0x8000000000000000ULL)) {
 		dprintf(LOG_INFO, FNAME, "detected a singular point");
 		return (1);
 	}
