Index: wide-dhcpv6/dhcp6s.c
===================================================================
--- wide-dhcpv6.orig/dhcp6s.c	2008-02-19 10:37:33.000000000 +0100
+++ wide-dhcpv6/dhcp6s.c	2008-02-19 10:38:00.000000000 +0100
@@ -2503,7 +2503,7 @@
 					saddr.pltime =
 					    blv->val_statefuladdr6.pltime;
 					saddr.vltime =
-					    blv->val_statefuladdr6.pltime;
+					    blv->val_statefuladdr6.vltime;
 				}
 
 				if (dhcp6_add_listval(&ialist,
