Index: wide-dhcpv6/common.c
===================================================================
--- wide-dhcpv6.orig/common.c	2008-02-19 10:44:07.000000000 +0100
+++ wide-dhcpv6/common.c	2008-02-19 10:45:10.000000000 +0100
@@ -514,7 +514,7 @@
 		memcpy(cp, name, nlen);
 		cp += nlen;
 	}
-	if (copy_option(type, optlen, tmpbuf, p, optep, len) != 0) {
+	if (copy_option(type, cp - tmpbuf, tmpbuf, p, optep, len) != 0) {
 		free(tmpbuf);
 		return -1;
 	}
